check-version.sh

```diff
@@ -0,0 +1,96 @@
+#!/bin/bash
+
+set -u
+
+if [ -z ${1:-} ]; then
+    echo "ERROR: Usage: $0 <directory>" >&2
+    exit 1
+fi
+dir=${1:-}
+
+if [ ! -d $dir ]; then
+    echo "ERROR: Non-existing directory '$dir' specified" >&2
+    exit 2
+fi
+
+# determine current version & checksum for specified directory
+curr_version_line=$(grep '^\\LARGE Version' $dir/*.tex)
+curr_version_file=$(echo $curr_version_line | cut -f1 -d:)
+curr_version=$(echo $curr_version_line | cut -f2 -d: | cut -f3 -d' ')
+if [ -z $curr_version ]; then
+    echo "ERROR: Failed to determine current version for '$dir'" >&2
+    exit 3
+else
+    echo "Current version for $dir: $curr_version (found in $curr_version_file)"
+fi
+
+travis_label=$(echo $dir | tr '[[:lower:]]' '[[:upper:]]' | tr '-' '_')
+curr_checksum=$(grep "${travis_label}_EXPECTED=" .travis.yml | sed 's/.* \(.*\)"$/\1/g')
+if [ -z $curr_checksum ]; then
+    echo "ERROR: Failed to determine current checksum for '$dir'" >&2
+    exit 4
+else
+    echo "Current checksum for $dir: $curr_checksum"
+fi
+
+# compute new checksum, to check whether current version/checksum is up-to-date
+for checksum_tool_cand in sha256sum shasum;
+do
+    which $checksum_tool_cand > /dev/null
+    if [ $? -eq 0 ]; then
+        checksum_tool=$checksum_tool_cand
+        break
+    fi
+done
+if [ "x$checksum_tool" == "xshasum" ]; then
+    checksum_tool="shasum -a 256"
+fi
+
+if [ -z $checksum_tool ]; then
+    echo "ERROR: No checksum tool found!" >&2
+    exit 4
+fi
+
+mkdir -p /tmp/$USER
```

Comment:
euh, why this? Isn't `mktemp` sufficient?

Reply:
force of habit, will fix